common/src/main/java/de/michelinside/glucodatahandler/common/notification/Vibrator.kt

```diff
@@ -38,11 +39,24 @@ object Vibrator {
                 else
                     amplitudePatterns[i] = 0
             }
-            vibrator.vibrate(VibrationEffect.createWaveform(pattern, amplitudePatterns, repeat))
+            VibrationEffect.createWaveform(pattern, amplitudePatterns, repeat)
         } else {
             Log.d(LOG_ID, "Vibrate for $duration ms")
-            vibrator.vibrate(VibrationEffect.createWaveform(pattern, repeat))
+            VibrationEffect.createWaveform(pattern, repeat)
         }
+
+        try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                // Use new API for Android 12+
+                val vibratorManager = GlucoDataService.context!!.getSystemService(Context.VIBRATOR_MANAGER_SERVICE) as VibratorManager
```

Comment:
vibrator instance is already the default vibrator of VibratorManager!? Also cancel will not work!?
So I think, we will need something more.... 
I will test.